wordpress-mobile/WordPress-Android
# Add post publish QS task

Fixes #13845

This PR adds logic to show the quick start focus point on the FAB from the improved my site fragment. I was considering moving the FAB to the fragment (because I think it logically belongs there) but the changes I'd have to make were massive and the code with two access point gets really complicated. In the end the code shows the FAB from the fragment but handles the logic to hide as before - through the view model which now calls QuickStartRepository when the flag is enabled.

![publish_post_video](https://user-images.githubusercontent.com/1079756/106489841-7cc3d880-64b5-11eb-96da-18470dd43516.gif)


To test:
- Turn on the MySiteImprovements flag and restart the app
- Make sure you are using a new site with Quick Start in progress
- Click on "Publish post" QS card
- Notice a Snackbar with a correct message
- Notice a focus point on the FAB
- Click on another QS card ("Update title")
- Notice the focus point is gone
- Click on the "Publish post" QS card again
- Click on the FAB
- Notice the focus point on the "Blog post" item
- Click on the "Blog post" item 
- Go back
- Notice the "Publish post" card is now gone

PR submission checklist:
